index.js

```diff
@@ -0,0 +1,565 @@
+
+  alexa.registerHandlers(
+    handlers,
+    start_handlers,
+    lection_handlers,
+    music_handlers,
+    game_handlers
+  );
+
+  alexa.appId = appId;
+  alexa.execute();
+});
+
+let server = app.listen(process.env.PORT || 4567, function () {
+  let host = server.address().address;
+  let port = server.address().port;
+  console.log("Server listening at http://%s:%s", host, port);
+  console.log("");
+});
+
+const handlers = {
+  "LaunchRequest": function () {
+    db.addUser(this.event.session.user.userId);
+
+    // USE ONCE AT DATABASE CREATION
+    //hf.createAllLectures();
+
+    this.handler.state = "_START";
+    this.emitWithState("Start");
+  },
+  "Unhandled": function () {
+    //this.response.speak("Auf Wiedersehen!").audioPlayerStop();
+    //this.emit(":responseReady");
+    this.handler.state = "_START";
+    this.emitWithState("Start");
+  }
+};
+
+const start_handlers = Alexa.CreateStateHandler("_START", {
+  "Start": function () {
+    if(this.attributes["unhandled"]){
+      this.attributes["unhandled"] = false;
+      this.response.speak("Das hat leider nicht geklappt, versuchen wir es noch einmal. Möchtest du lernen, üben oder Lieder hören?")
+      .listen("Sage lernen, üben oder Lieder hören").audioPlayerStop();
+      this.emit(":responseReady");
+    } else {
+      this.attributes["unhandled"] = false;
+      this.response.speak("Möchtest du lernen, üben oder Lieder hören?")
+      .listen("Sage lernen, üben oder Lieder hören").audioPlayerStop();
+      this.emit(":responseReady");
+    }
+  },
+  'LearnIntent': function () {
+    this.handler.state = "_LECTION";
+    this.emitWithState("Lection");
+  },
+  'GameReadyIntent': function(){
+    this.handler.state = "_GAME";
+    this.attributes["game"] = true;
+    this.emitWithState("Game");
+  },
+  "SongIntent": function () {
+    this.handler.state = "_MUSIC";
+    this.emitWithState("Music");
+  },
+  "HomeIntent": function () {
+    this.handler.state = "_START";
+    this.emitWithState("Start");
+  },
+  "AMAZON.StopIntent": function () {
+    this.response.speak('Auf Wiederhören!');
+    this.emit(":responseReady");
+  },
+  "AMAZON.CancelIntent": function () {
+    this.response.speak('Auf Wiederhören!');
+    this.emit(":responseReady");
+  },
+  "AMAZON.HelpIntent": function () {
+    this.response.speak("Sage lernen, üben oder Lieder hören").listen("Sage lernen, üben oder Lieder hören");
+    this.emit(":responseReady");
+  },
+  "Unhandled": function () {
+    this.attributes["unhandled"] = true;
+    this.emitWithState("Start");
+  }
+});
+
+const music_handlers = Alexa.CreateStateHandler("_MUSIC", {
+  "Music": function () {
+    this.emitWithState("MusicIntent");
+  },
+  "MusicIntent": function () {
+    db.getPointsFromUser(this.event.session.user.userId, (points) => {
+      if(points >= 1){
+        db.removeSongPointsFromUser(this.event.session.user.userId);
+
+        let r = hf.getRandomNumber(3);
```

Comment:
Number of songs should not be hard-coded

Reply:
And the paths down below should not be hard-coded either.

Reply:
this was a last minute decision to make this version functional, I will replace hard coded number by variables